web/src/engine/websites/MangaReaderTo.ts

```diff
@@ -0,0 +1,282 @@
+    const title = element instanceof HTMLAnchorElement ? element.title.replace(/([^:]*):(.*)/, (match, g1, g2) => g1.trim().toLowerCase() === g2.trim().toLowerCase() ? g1 : match).trim() : element.textContent.trim();
+    const lang = link.match(/\/read\/[^/]+\/([^/]+)/)[1];
+    return {
+        id: link,
+        title: `${title} (${lang})`,
+    };
+
+}
+
+@Common.MangaCSS(/^{origin}\/[^/]+-\d+$/, 'div#ani_detail div.anisc-detail h2.manga-name')
+@Common.MangasMultiPageCSS('/az-list?page={page}', '#main-content div.manga-detail h3 a', 1, 1, 0, Common.AnchorInfoExtractor(true))
+@Common.ChaptersSinglePageCSS('div.chapters-list-ul ul li a, div.volume-list-ul div.manga-poster', ChapterExtractor)
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangareaderto', `MangaReader.to`, 'https://mangareader.to', Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.Multilingual, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const chapterurl = new URL(chapter.Identifier, this.URI);
+        const [data] = await FetchCSS(new Request(chapterurl), 'div#wrapper');
+        const readingId = data.dataset.readingId;
+
+        const uri = new URL(`ajax/image/list/${chapterurl.href.includes('chapter') ? 'chap' : 'vol'}/${readingId}?quality=high`, this.URI);
+        const { status, html } = await FetchJSON<AJAXResponse>(new Request(uri));
+        if (!status) return [];
+
+        const dom = new DOMParser().parseFromString(html, 'text/html');
+        const imagesArr = Array.from(dom.querySelectorAll<HTMLDivElement>('div.iv-card'));
+        return imagesArr.map(image => new Page(this, chapter, new URL(image.dataset.url, this.URI), { shuffled: [...image.classList].includes('shuffled') }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        let blob = await Common.FetchImageAjax.call(this, page, priority, signal);
+        return !page.Parameters['shuffled'] ? blob : DeScramble(blob, async (image, ctx) => {
+
+            const pieceDimension = 200;
+            const defaultSeed = 'stay';
+            const numpieces = Math.ceil(image.width / pieceDimension) * Math.ceil(image.height / pieceDimension);
+            const numcol = Math.ceil(image.width / pieceDimension);
+            const slicesMap = new Map<string, TSlice[]>();
+            for (let index = 0; index < numpieces; index++) {
+                const g = Math.trunc(index / numcol);
+                const slice: TSlice = { x: (index - g * numcol) * pieceDimension, y: g * pieceDimension, width: undefined, height: undefined };
+                slice.width = pieceDimension - (slice.x + pieceDimension <= image.width ? 0 : slice.x + pieceDimension - image.width);
```

Comment:
Simplify the whole expression with`Math.min` or `Math.max` for better readability.
I think it is same as `Math.min(image.width - slice.x, pieceDimension)`.
Same for height calculation.

Reply:
Nice catch!